packages/yew/src/dom_bundle/dom_position.rs

```diff
@@ -0,0 +1,254 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use web_sys::{Element, Node};
+
+/// A position in the list of children of an implicit parent [`Element`].
+///
+/// This can either be in front of a `DomSlot::at(next_sibling)`, at the end of the list with
+/// `DomSlot::at_end()`, or a dynamic position in the list with [`DynamicDomSlot::to_position`].
+#[derive(Clone)]
+pub struct DomSlot {
+    variant: DomSlotVariant,
+}
+
+#[derive(Clone)]
+enum DomSlotVariant {
+    Node(Option<Node>),
+    Chained(DynamicDomSlot),
+}
+
+/// A dynamic dom slot can be retargeted. This change is also seen by the [`DomSlot`] from
+/// [`Self::to_position`] before the retargeting took place.
+#[derive(Clone)]
+pub struct DynamicDomSlot {
```

Comment:
```suggestion
pub(crate) struct DynamicDomSlot {
```

It's better to explicitly declare this as crate-private so we don't leak these types accidentally.